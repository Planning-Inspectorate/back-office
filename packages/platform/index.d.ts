import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
