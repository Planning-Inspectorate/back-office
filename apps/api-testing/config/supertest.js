// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
