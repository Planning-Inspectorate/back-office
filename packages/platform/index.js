export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
