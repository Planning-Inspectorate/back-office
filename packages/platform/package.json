{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
