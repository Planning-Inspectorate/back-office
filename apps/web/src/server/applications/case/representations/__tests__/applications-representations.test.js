import { jest } from '@jest/globals';
import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { createTestEnvironment } from '../../../../../../testing/index.js';
import { representationsFixture } from '../__fixtures__/representations.fixture.js';
import dateUtils from '../../../../lib/add-business-days-to-date.js';

const { app, installMockApi, teardown } = createTestEnvironment();
const request = supertest(app);

const mockReviewDate = '2023-01-20';

jest
	.spyOn(dateUtils, 'addBusinessDaysToDate')
	.mockImplementation(() => Promise.resolve(new Date(mockReviewDate)));

const mockCaseReference = {
	title: 'mock title',
	status: 'in test',
	reference: 'mock reference',
	keyDates: {
		preExamination: {
			dateOfRelevantRepresentationClose: '1672531200'
		}
	},
	sector: {
		name: 'mock_sector',
		displayNameEn: 'Mock Sector Display Name'
	},
	subSector: {
		name: 'mock_sub_sector',
		displayNameEn: 'Mock Sub-Sector Display Name'
	}
};
const nocks = () => {
	nock('http://test/').get('/applications/1').times(2).reply(200, mockCaseReference);
	nock('http://test/')
		.get(`/applications/1/representations`)
		.query({
			searchTerm: 'mock-search-term',
			sortBy: '',
			page: 1,
			pageSize: 25,
			status: '',
			under18: false,
			withAttachment: false
		})
		.reply(200, {
			page: 1,
			pageSize: 25,
			pageCount: 1,
			itemCount: 3,
			items: [representationsFixture.items[0]],
			filters: [
				{ name: 'PUBLISHED', count: 1 },
				{ name: 'AWAITING_REVIEW', count: 1 },
				{ name: 'VALID', count: 1 }
			]
		})
		.persist();
	nock('http://test/')
		.get(`/applications/1/representations`)
		.query({
			searchTerm: '',
			sortBy: '',
			status: '',
			page: 1,
			pageSize: 25,
			under18: false,
			withAttachment: false
		})
		.reply(200, {
			...representationsFixture,
			filters: [
				{ name: 'PUBLISHED', count: 1 },
				{ name: 'AWAITING_REVIEW', count: 1 },
				{ name: 'VALID', count: 1 }
			]
		})
		.persist();
	nock('http://test/')
		.get('/applications/1/representations/publishable')
		.reply(200, { previouslyPublished: true, itemCount: 1 });
	nock('http://test/')
		.get('/applications/1/representations')
		.query({
			status: 'PUBLISHED'
		})
		.reply(200, { itemCount: 1, items: [representationsFixture.items[2]] });
};

describe('applications representations', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	afterAll(() => {
		nock.cleanAll();
	});

	const baseUrl = '/applications-service/case/1/relevant-representations';

	describe('GET /applications-service/:caseId/relevant-representations', () => {
		beforeEach(async () => {
			nocks();
			await request.get('/applications-service/');
		});

		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			// Assert - case ref data is present
			expect(element.innerHTML).toContain(mockCaseReference.title);
			expect(element.innerHTML).toContain(mockCaseReference.status);
			expect(element.innerHTML).toContain(mockCaseReference.reference);

			// Assert - Add a representation button is present
			expect(element.innerHTML).toContain('Add a representation');

			// Assert - css classes applied to the status
			expect(element.innerHTML).toContain(
				`<strong class="govuk-tag govuk-tag--grey" id="list-convictions-status-1">Awaiting review</strong>`
			);
			expect(element.innerHTML).toContain(
				`<strong class="govuk-tag govuk-tag--green" id="list-convictions-status-2">Valid</strong>`
			);
			expect(element.innerHTML).toContain(
				`<strong class="govuk-tag govuk-tag--green" id="list-convictions-status-3">Published</strong>`
			);
		});
	});

	describe('GET /applications-service/:id/relevant-representations with a search term', () => {
		beforeEach(async () => {
			nocks();
			await request.get('/applications-service/');
		});

		it('should render the page with search term', async () => {
			const response = await request.get(`${baseUrl}?searchTerm=mock-search-term`);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			// Assert - case ref data is present
			expect(element.innerHTML).toContain(mockCaseReference.title);
			expect(element.innerHTML).toContain(mockCaseReference.status);
			expect(element.innerHTML).toContain(mockCaseReference.reference);

			// Assert - css classes applied to the status
			expect(element.innerHTML).toContain(
				`<strong class="govuk-tag govuk-tag--grey" id="list-convictions-status-1">Awaiting review</strong>`
			);
		});
	});
});
