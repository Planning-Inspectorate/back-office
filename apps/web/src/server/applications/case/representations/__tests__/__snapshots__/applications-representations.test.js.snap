// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications representations GET /applications-service/:caseId/relevant-representations should render the page 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="top-errors-hook" role="alert"></div>
    <div class="govuk-grid-row govuk-!-padding-top-9 govuk-!-padding-bottom-2 govuk-!-margin-left-0 govuk-!-margin-bottom-7 pins-masthead-summary">
        <div class="govuk-grid-column-full">
            <h4 class="govuk-heading-m"> mock reference </h4>
            <h1 class="govuk-heading-xl">mock title</h1>
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full govuk-body">,
                    <br>
                    <br><strong class="govuk-tag govuk-tag-- pins-tag--no-max-width">in test</strong>
                </div>
            </div>
        </div>
    </div>
    <div class="govuk-notification-banner" role="region" aria-labelledby="govuk-notification-banner-title"
    data-module="govuk-notification-banner" id="unpublished-changes-banner">
        <div class="govuk-notification-banner__header">
            <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title"> Important</h2>
        </div>
        <div class="govuk-notification-banner__content">
            <p class="govuk-notification-banner__heading">Changes have been made. These changes have not been published.</p>
            <p class="govuk-body"><a class="govuk-notification-banner__link" href="/applications-service/case/1/relevant-representations/select-representations-for-publishing"> Preview and publish the changes</a>
            </p><a href="#" id="clear-update-notification-link" class="govuk-link">Clear notification</a>
        </div>
    </div>
    <script nonce="EdcOUaJ8lczj9tIPO0lPow==">
        document.getElementById('clear-update-notification-link').addEventListener('click', (e) => {
        				e.preventDefault()
        				document.getElementById('unpublished-changes-banner').style.display = 'none'
        			});
    </script>
    <h2 class="govuk-heading-l">Relevant representations</h2>
    <form method="GET" action="/applications-service/case/1/relevant-representations">
        <div class="govuk-grid-row">
            <div class="govuk-grid-column-two-thirds">
                <div class="pins-masthead-summary govuk-!-padding-4 govuk-!-margin-bottom-4">
                    <div class="govuk-form-group">
                        <div class="filter-box">
                            <div class="govuk-form-group">
                                <fieldset class="govuk-fieldset">
                                    <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">Filter</legend>
                                    <div class="govuk-checkboxes govuk-checkboxes--small" data-module="govuk-checkboxes">
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters" name="filters" type="checkbox"
                                            value="AWAITING_REVIEW">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters">Awaiting review (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-2" name="filters" type="checkbox"
                                            value="VALID">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-2">Valid (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-3" name="filters" type="checkbox"
                                            value="DRAFT">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-3">Draft (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-4" name="filters" type="checkbox"
                                            value="PUBLISHED">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-4">Published (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-5" name="filters" type="checkbox"
                                            value="REFERRED">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-5">Referred (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-6" name="filters" type="checkbox"
                                            value="WITHDRAWN">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-6">Withdrawn (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-7" name="filters" type="checkbox"
                                            value="INVALID">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-7">Invalid (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-8" name="filters" type="checkbox"
                                            value="UNDER_18">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-8">Under 18 (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-9" name="filters" type="checkbox"
                                            value="WITH_ATTACHMENT">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-9">With attachment (3)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-10" name="filters"
                                            type="checkbox" value="UNPUBLISHED">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-10">Unpublished (0)</label>
                                        </div>
                                    </div>
                                </fieldset>
                            </div>
                        </div>
                    </div>
                    <div class="govuk-button-group govuk-!-margin-top-0">
                        <button class="govuk-button" id="toggleButton" data-module="govuk-button">Apply filters</button><a href="/applications-service/case/1/relevant-representations"
                        class="govuk-link govuk-link--no-visited-state"> Clear filters</a>
                    </div>
                </div>
            </div>
            <div class="govuk-grid-column-one-third">
                <h3 class="govuk-heading-m govuk-!-margin-bottom-6"> Key dates</h3>
                <p class="govuk-body govuk-!-margin-bottom-1"><strong>1 January 2023</strong>
                </p>
                <p class="govuk-body govuk-!-margin-bottom-6">Closed date</p>
                <p class="govuk-body govuk-!-margin-bottom-1"><strong> 20 January 2023</strong>
                </p>
                <p class="govuk-body govuk-!-margin-bottom-6">Target date for review and completion</p><a href="/applications-service/case/1/relevant-representations/contact-details?repType=represented"
                role="button" draggable="false" class="govuk-button govuk-button--secondary"
                data-module="govuk-button"> Add a representation</a>
                <p class="govuk-body"><a class="govuk-link" href="/applications-service/case/1/relevant-representations/select-representations-for-publishing">View publishing queue</a>
                </p>
                <p class="govuk-body"><a class="govuk-link" href="/applications-service/case/1/relevant-representations/api/download/published"
                    download> Download mailing list (CSV)</a>
                </p>
                <p class="govuk-body"><a class="govuk-link" href="/applications-service/case/1/relevant-representations/api/download/valid"
                    download> Download valid representations (CSV)</a>
                </p>
            </div>
        </div>
        <h3 class="govuk-heading-m">Search representations</h3>
        <input type="hidden" name="page" value="1">
        <input type="hidden" name="pageSize" value="25">
        <input type="hidden" name="sortBy" value="">
        <div class="govuk-grid-row">
            <div class="govuk-grid-column-one-half">
                <div class="govuk-form-group">
                    <input class="govuk-input" id="searchTerm" name="searchTerm" type="text">
                </div>
            </div>
            <div class="govuk-grid-column-one-half">
                <button value="updated" type="submit" name="searchUpdated" class="govuk-button govuk-button--secondary"
                data-module="govuk-button">Search</button>
            </div>
        </div>
        <div class="govuk-grid-row">
            <div class="govuk-grid-column-one-half">
                <p>Showing<b>1</b> to <b>3</b> of <b>3</b> results</p>
            </div>
            <div class="govuk-grid-column-one-half text-align--right">
                <p><span class="govuk-!-margin-right-2"><strong>Results per page</strong></span>
                    <span                     class="govuk-!-margin-left-1"><a class="govuk-link" href="relevant-representations?page=1&amp;pageSize=25"><span class="govuk-visually-hidden">View </span>25<span class="govuk-visually-hidden"> results per page</span></a>
                        </span><span class="govuk-!-margin-left-1">|</span><span class="govuk-!-margin-left-1"><a class="govuk-link" href="relevant-representations?page=1&amp;pageSize=50"><span class="govuk-visually-hidden">View </span>50
                        <span                         class="govuk-visually-hidden">results per page</span>
                            </a>
                            </span><span class="govuk-!-margin-left-1">|</span><span class="govuk-!-margin-left-1"><a class="govuk-link" href="relevant-representations?page=1&amp;pageSize=100"><span class="govuk-visually-hidden">View </span>100
                            <span                             class="govuk-visually-hidden">results per page</span>
                                </a>
                                </span>
                </p>
            </div>
        </div>
        <table id="rr-table" class="govuk-table pins-table--representations">
            <thead class="govuk-table__head">
                <tr class="govuk-table__row">
                    <th scope="col" class="govuk-table__header"><a href="relevant-representations?page=1&amp;sortBy=reference#rr-table"
                        class="sort-table__link "> Reference<span class="sort-table__link-icon  sort-table__link-icon--small "></span></a>
                    </th>
                    <th scope="col" class="govuk-table__header"><a href="relevant-representations?page=1&amp;sortBy=displayName#rr-table"
                        class="sort-table__link "> From<span class="sort-table__link-icon  sort-table__link-icon--small "></span></a>
                    </th>
                    <th scope="col" class="govuk-table__header"><a href="relevant-representations?page=1&amp;sortBy=received#rr-table"
                        class="sort-table__link "> Date received<span class="sort-table__link-icon  sort-table__link-icon--small "></span></a>
                    </th>
                    <th scope="col" class="govuk-table__header"><a href="relevant-representations?page=1&amp;sortBy=redacted#rr-table"
                        class="sort-table__link "> Redacted<span class="sort-table__link-icon  sort-table__link-icon--small "></span></a>
                    </th>
                    <th scope="col" class="govuk-table__header"><a href="relevant-representations?page=1&amp;sortBy=status#rr-table" class="sort-table__link "> Status<span class="sort-table__link-icon  sort-table__link-icon--small "></span></a>
                    </th>
                    <th scope="col" class="govuk-table__header">Action</th>
                </tr>
            </thead>
            <tbody class="govuk-table__body">
                <tr class="govuk-table__row">
                    <td class="govuk-table__cell">mock reference</td>
                    <td class="govuk-table__cell">org name 1</td>
                    <td class="govuk-table__cell">01 Jan 2022</td>
                    <td class="govuk-table__cell">Redacted</td>
                    <td class="govuk-table__cell"><strong class="govuk-tag govuk-tag--grey" id="list-convictions-status-1">Awaiting review</strong>
                    </td>
                    <td class="govuk-table__cell"><a class="govuk-link" href="/applications-service/case/1/relevant-representations/1/representation-details?repMode=summary"> Review <span class="govuk-visually-hidden">org name 1</span></a>
                    </td>
                </tr>
                <tr class="govuk-table__row">
                    <td class="govuk-table__cell">mock reference</td>
                    <td class="govuk-table__cell">first lastName</td>
                    <td class="govuk-table__cell">01 Jan 2022</td>
                    <td class="govuk-table__cell">Unredacted</td>
                    <td class="govuk-table__cell"><strong class="govuk-tag govuk-tag--green" id="list-convictions-status-2">Valid</strong>
                    </td>
                    <td class="govuk-table__cell"><a class="govuk-link" href="/applications-service/case/1/relevant-representations/2/representation-details?repMode=summary"> Review <span class="govuk-visually-hidden">first lastName</span></a>
                    </td>
                </tr>
                <tr class="govuk-table__row">
                    <td class="govuk-table__cell">mock reference</td>
                    <td class="govuk-table__cell">org name 2</td>
                    <td class="govuk-table__cell">01 Jan 2022</td>
                    <td class="govuk-table__cell">Unredacted</td>
                    <td class="govuk-table__cell"><strong class="govuk-tag govuk-tag--green" id="list-convictions-status-3">Published</strong>
                    </td>
                    <td class="govuk-table__cell"><a class="govuk-link" href="/applications-service/case/1/relevant-representations/3/representation-details?repMode=summary"> Review <span class="govuk-visually-hidden">org name 2</span></a>
                    </td>
                </tr>
            </tbody>
        </table>
    </form>
</main>"
`;

exports[`applications representations GET /applications-service/:id/relevant-representations with a search term should render the page with search term 1`] = `
"<main class="govuk-main-wrapper" id="main-content">
    <div class="top-errors-hook" role="alert"></div>
    <div class="govuk-grid-row govuk-!-padding-top-9 govuk-!-padding-bottom-2 govuk-!-margin-left-0 govuk-!-margin-bottom-7 pins-masthead-summary">
        <div class="govuk-grid-column-full">
            <h4 class="govuk-heading-m"> mock reference </h4>
            <h1 class="govuk-heading-xl">mock title</h1>
            <div class="govuk-grid-row">
                <div class="govuk-grid-column-full govuk-body">,
                    <br>
                    <br><strong class="govuk-tag govuk-tag-- pins-tag--no-max-width">in test</strong>
                </div>
            </div>
        </div>
    </div>
    <div class="govuk-notification-banner" role="region" aria-labelledby="govuk-notification-banner-title"
    data-module="govuk-notification-banner" id="unpublished-changes-banner">
        <div class="govuk-notification-banner__header">
            <h2 class="govuk-notification-banner__title" id="govuk-notification-banner-title"> Important</h2>
        </div>
        <div class="govuk-notification-banner__content">
            <p class="govuk-notification-banner__heading">Changes have been made. These changes have not been published.</p>
            <p class="govuk-body"><a class="govuk-notification-banner__link" href="/applications-service/case/1/relevant-representations/select-representations-for-publishing"> Preview and publish the changes</a>
            </p><a href="#" id="clear-update-notification-link" class="govuk-link">Clear notification</a>
        </div>
    </div>
    <script nonce="EdcOUaJ8lczj9tIPO0lPow==">
        document.getElementById('clear-update-notification-link').addEventListener('click', (e) => {
        				e.preventDefault()
        				document.getElementById('unpublished-changes-banner').style.display = 'none'
        			});
    </script>
    <h2 class="govuk-heading-l">Relevant representations</h2>
    <form method="GET" action="/applications-service/case/1/relevant-representations">
        <div class="govuk-grid-row">
            <div class="govuk-grid-column-two-thirds">
                <div class="pins-masthead-summary govuk-!-padding-4 govuk-!-margin-bottom-4">
                    <div class="govuk-form-group">
                        <div class="filter-box">
                            <div class="govuk-form-group">
                                <fieldset class="govuk-fieldset">
                                    <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">Filter</legend>
                                    <div class="govuk-checkboxes govuk-checkboxes--small" data-module="govuk-checkboxes">
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters" name="filters" type="checkbox"
                                            value="AWAITING_REVIEW">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters">Awaiting review (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-2" name="filters" type="checkbox"
                                            value="VALID">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-2">Valid (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-3" name="filters" type="checkbox"
                                            value="DRAFT">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-3">Draft (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-4" name="filters" type="checkbox"
                                            value="PUBLISHED">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-4">Published (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-5" name="filters" type="checkbox"
                                            value="REFERRED">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-5">Referred (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-6" name="filters" type="checkbox"
                                            value="WITHDRAWN">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-6">Withdrawn (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-7" name="filters" type="checkbox"
                                            value="INVALID">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-7">Invalid (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-8" name="filters" type="checkbox"
                                            value="UNDER_18">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-8">Under 18 (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-9" name="filters" type="checkbox"
                                            value="WITH_ATTACHMENT">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-9">With attachment (0)</label>
                                        </div>
                                        <div class="govuk-checkboxes__item">
                                            <input class="govuk-checkboxes__input" id="filters-10" name="filters"
                                            type="checkbox" value="UNPUBLISHED">
                                            <label class="govuk-label govuk-checkboxes__label" for="filters-10">Unpublished (0)</label>
                                        </div>
                                    </div>
                                </fieldset>
                            </div>
                        </div>
                    </div>
                    <div class="govuk-button-group govuk-!-margin-top-0">
                        <button class="govuk-button" id="toggleButton" data-module="govuk-button">Apply filters</button><a href="/applications-service/case/1/relevant-representations"
                        class="govuk-link govuk-link--no-visited-state"> Clear filters</a>
                    </div>
                </div>
            </div>
            <div class="govuk-grid-column-one-third">
                <h3 class="govuk-heading-m govuk-!-margin-bottom-6"> Key dates</h3>
                <p class="govuk-body govuk-!-margin-bottom-1"><strong>1 January 2023</strong>
                </p>
                <p class="govuk-body govuk-!-margin-bottom-6">Closed date</p>
                <p class="govuk-body govuk-!-margin-bottom-1"><strong> 20 January 2023</strong>
                </p>
                <p class="govuk-body govuk-!-margin-bottom-6">Target date for review and completion</p><a href="/applications-service/case/1/relevant-representations/contact-details?repType=represented"
                role="button" draggable="false" class="govuk-button govuk-button--secondary"
                data-module="govuk-button"> Add a representation</a>
                <p class="govuk-body"><a class="govuk-link" href="/applications-service/case/1/relevant-representations/select-representations-for-publishing">View publishing queue</a>
                </p>
                <p class="govuk-body"><a class="govuk-link" href="/applications-service/case/1/relevant-representations/api/download/published"
                    download> Download mailing list (CSV)</a>
                </p>
                <p class="govuk-body"><a class="govuk-link" href="/applications-service/case/1/relevant-representations/api/download/valid"
                    download> Download valid representations (CSV)</a>
                </p>
            </div>
        </div>
        <h3 class="govuk-heading-m">Search representations</h3>
        <input type="hidden" name="page" value="1">
        <input type="hidden" name="pageSize" value="25">
        <input type="hidden" name="sortBy" value="">
        <div class="govuk-grid-row">
            <div class="govuk-grid-column-one-half">
                <div class="govuk-form-group">
                    <input class="govuk-input" id="searchTerm" name="searchTerm" type="text"
                    value="mock-search-term">
                </div>
            </div>
            <div class="govuk-grid-column-one-half">
                <button value="updated" type="submit" name="searchUpdated" class="govuk-button govuk-button--secondary"
                data-module="govuk-button">Search</button>
            </div>
        </div>
        <div class="govuk-grid-row">
            <div class="govuk-grid-column-one-half">
                <p>Showing<b>1</b> to <b>3</b> of <b>3</b> results</p>
            </div>
            <div class="govuk-grid-column-one-half text-align--right">
                <p><span class="govuk-!-margin-right-2"><strong>Results per page</strong></span>
                    <span                     class="govuk-!-margin-left-1"><a class="govuk-link" href="relevant-representations?searchTerm=mock-search-term&amp;page=1&amp;pageSize=25"><span class="govuk-visually-hidden">View </span>25<span class="govuk-visually-hidden"> results per page</span></a>
                        </span><span class="govuk-!-margin-left-1">|</span><span class="govuk-!-margin-left-1"><a class="govuk-link" href="relevant-representations?searchTerm=mock-search-term&amp;page=1&amp;pageSize=50"><span class="govuk-visually-hidden">View </span>50
                        <span                         class="govuk-visually-hidden">results per page</span>
                            </a>
                            </span><span class="govuk-!-margin-left-1">|</span><span class="govuk-!-margin-left-1"><a class="govuk-link" href="relevant-representations?searchTerm=mock-search-term&amp;page=1&amp;pageSize=100"><span class="govuk-visually-hidden">View </span>100
                            <span                             class="govuk-visually-hidden">results per page</span>
                                </a>
                                </span>
                </p>
            </div>
        </div>
        <table id="rr-table" class="govuk-table pins-table--representations">
            <thead class="govuk-table__head">
                <tr class="govuk-table__row">
                    <th scope="col" class="govuk-table__header"><a href="relevant-representations?searchTerm=mock-search-term&amp;page=1&amp;sortBy=reference#rr-table"
                        class="sort-table__link "> Reference<span class="sort-table__link-icon  sort-table__link-icon--small "></span></a>
                    </th>
                    <th scope="col" class="govuk-table__header"><a href="relevant-representations?searchTerm=mock-search-term&amp;page=1&amp;sortBy=displayName#rr-table"
                        class="sort-table__link "> From<span class="sort-table__link-icon  sort-table__link-icon--small "></span></a>
                    </th>
                    <th scope="col" class="govuk-table__header"><a href="relevant-representations?searchTerm=mock-search-term&amp;page=1&amp;sortBy=received#rr-table"
                        class="sort-table__link "> Date received<span class="sort-table__link-icon  sort-table__link-icon--small "></span></a>
                    </th>
                    <th scope="col" class="govuk-table__header"><a href="relevant-representations?searchTerm=mock-search-term&amp;page=1&amp;sortBy=redacted#rr-table"
                        class="sort-table__link "> Redacted<span class="sort-table__link-icon  sort-table__link-icon--small "></span></a>
                    </th>
                    <th scope="col" class="govuk-table__header"><a href="relevant-representations?searchTerm=mock-search-term&amp;page=1&amp;sortBy=status#rr-table"
                        class="sort-table__link "> Status<span class="sort-table__link-icon  sort-table__link-icon--small "></span></a>
                    </th>
                    <th scope="col" class="govuk-table__header">Action</th>
                </tr>
            </thead>
            <tbody class="govuk-table__body">
                <tr class="govuk-table__row">
                    <td class="govuk-table__cell">mock reference</td>
                    <td class="govuk-table__cell">org name 1</td>
                    <td class="govuk-table__cell">01 Jan 2022</td>
                    <td class="govuk-table__cell">Redacted</td>
                    <td class="govuk-table__cell"><strong class="govuk-tag govuk-tag--grey" id="list-convictions-status-1">Awaiting review</strong>
                    </td>
                    <td class="govuk-table__cell"><a class="govuk-link" href="/applications-service/case/1/relevant-representations/1/representation-details?repMode=summary"> Review <span class="govuk-visually-hidden">org name 1</span></a>
                    </td>
                </tr>
            </tbody>
        </table>
    </form>
</main>"
`;
